packages/graphql/tests/integration/interfaces-top-level.int.test.ts

```diff
@@ -176,6 +183,108 @@ describe("Interfaces tests", () => {
                         id: "2",
                     },
                 },
+                {
+                    id: "10",
+                    other: {
+                        id: "2",
+                    },
+                },
+            ],
+        });
+    });
+
+    test("should return results on top-level simple query on simple interface with filters", async () => {
+        const neoGraphql = new Neo4jGraphQL({
+            typeDefs,
+            driver,
+            features: {
+                authorization: {
+                    key: secret,
+                },
+            },
+            experimental: true,
+        });
+        schema = await neoGraphql.getSchema();
+
+        const query = `
+            query {
+                myOtherInterfaces(where: {_on:{ ${SomeNodeType}: { other: {id: "2"}} } }) {
+                    id
+                    ... on ${SomeNodeType} {
+                        id
+                        other {
+                            id
+                        }
+                    }
+                }
+            }
+        `;
+
+        const token = createBearerToken(secret, {});
+        const queryResult = await graphqlQuery(query, token);
+        expect(queryResult.errors).toBeUndefined();
+        expect(queryResult.data).toEqual({
+            myOtherInterfaces: [
+                {
+                    id: "1",
+                    other: {
+                        id: "2",
+                    },
+                },
+                {
+                    id: "10",
+                    other: {
+                        id: "2",
+                    },
+                },
+            ],
+        });
+    });
+
+    test("should return results on top-level simple query on interface target to a relationship with filters", async () => {
```

Comment:
Will be good to have some tests for the scenarios described in the test file
`packages/graphql/tests/tck/directives/interface-relationships/read.test.ts`:
- "Type filtering using onType"
- "Filter overriding using onType"

It will improve consistency with the current behavior from day 1!

Reply:
Added here, could you please check to confirm this is what you meant? https://github.com/neo4j/graphql/pull/4078/commits/c47da30ab335ac7e107b4472a98c248b3e6f1a76